modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlParserSelfTest.java

```diff
@@ -131,6 +131,17 @@ public void testTokenizationError() {
                 () -> IgniteSqlParser.parse("SELECT foo#bar", StatementParseResult.MODE));
     }
 
+    @Test
+    public void testIdentifier() {
+        IgniteSqlParser.parse("SELECT 1 as _foo", StatementParseResult.MODE);
+        IgniteSqlParser.parse("SELECT 1 as \"$foo\"", StatementParseResult.MODE);
```

Comment:
seems case with erroneous codePointAt need to be appended here ?

Reply:
What are erronless codepoints?

Reply:
ok - irrelevant, to check: 
`    public static boolean isValidNormalizedIdentifier(String identifier) {
        if (identifier == null || identifier.isEmpty()) {
            return false;
        }

        int codePoint = identifier.codePointAt(0);
        if (!identifierStart(codePoint)) {`